lib/scrub/scrub_runner.rb

```diff
@@ -0,0 +1,125 @@
+# A fully automated scrubber would:
+# Check org directories for new org-submitted files
+# Compare remote files to processed files
+# Copy new files to work directory
+# Validate and scrub files into loadable format
+# Load files into db
+# Upload report to org dir
+# ... should be able to do this for any number of orgs/files, at a whim or by cron,
+# ideally using a queue or pool of workers.
+
+require "data_sources/directory_locator"
+require "data_sources/ht_organizations"
+require "fileutils"
+require "loader/file_loader"
+require "loader/holding_loader"
+require "scrub/autoscrub"
+require "scrub/chunker"
+require "utils/file_transfer"
+
+# Example:
+# runner = Scrub::ScrubRunner.new(ORG)
+# runner.run
+
+module Scrub
+  # Scrubs and loads any new org-uploaded files.
+  class ScrubRunner
+    def initialize(organization)
+      @organization = organization
+      @ft = Utils::FileTransfer.new
+      validate
+    end
+
+    def validate
+      if Settings.local_member_data.nil?
+        raise "Need Settings.local_member_data to be set"
+      end
+      if Settings.remote_member_data.nil?
+        raise "Need Settings.remote_member_data to be set"
+      end
+      if @organization.nil?
+        raise "Need @organization to be set"
+      end
+    end
+
+    def run
+      puts "Running org #{@organization}."
+      new_files = check_new_files
+      puts "Found #{new_files.size} new files: #{new_files.join(", ")}."
+      new_files.each do |new_file|
+        run_file(new_file)
+      end
+    end
+
+    def run_file(file)
+      puts "Running file #{file}"
+      downloaded_file = download_to_work_dir(file)
+      scrubber = Scrub::AutoScrub.new(downloaded_file)
+      scrubber.run
+      scrubber.out_files.each do |scrubber_out_file|
+        puts "Ready to split #{scrubber_out_file} into chunks"
+        chunker = Scrub::Chunker.new(glob: scrubber_out_file, chunk_count: 4, out_ext: "ndj")
+        chunker.run
+        chunker.chunks.each do |chunk|
+          puts "Load chunk #{chunk}"
+          Loader::FileLoader.new(batch_loader: Loader::HoldingLoader.for(chunk)).load(chunk)
+        end
+        # do when chunks are done, except we don;t know how really
+        chunker.cleanup!
+        upload_to_member(scrubber.logger_path)
+      end
+    rescue
+      # If the scrub failed, remove the file from local storage, that we may try again.
+      FileUtils.rm(downloaded_file)
+      raise "Scrub failed, removing downloaded file #{downloaded_file}"
+    end
+
+    # Check org-uploaded files for any not previously seen files
+    def check_new_files
+      # Return new (as in not in old) files
+      remote_files = @ft.lsjson(remote_dir)
+      old_files = check_old_files
+
+      # Include in new_files only those remote_files whose name is not in old_files.
+      new_files = []
+      remote_files.each do |f|
```

Comment:
I think the intention is that `new_files` should contain anything that is:
- in `remote_files`
- does not end in `.log`
- is not in `old_files`

in which case you might be able to do something like:

```ruby
remote_files.reject { |f| f["Name"].end_with?(".log") } - old_files
```

That would compare more fields than just the name, but perhaps you could do:

```ruby
new_file_names = remote_files.map {|f| f["Name"] }.reject { |f| f.ends_with?(".log") } - old_files.map { |f| f["Name"] }
```

Reply:
I don't see how this improves upon or makes easier to read what's already there.

Reply:
I found it difficult to understand the intention of what `new_files` was doing without tracing the code (although it is noted in the spec). That said, I'm fine with leaving what we have here for the time being. Perhaps if we find that we need to change the logic here in the future, we could consider havingremote_files` return an array of objects instead of an array of hashes. Then we have some additional options for encapsulating the logic here and making the operations more clear.